usaspending_api/search/v2/views/spending_over_time.py

```diff
@@ -142,19 +150,67 @@ def parse_elasticsearch_bucket(self, bucket: dict) -> dict:
         It should be noted that `key_as_string` is the name given by `date_histogram` to represent the key
         for each bucket which is a date as a string.
         """
+
+        # extract the current time period
         key_as_date = datetime.strptime(bucket["key_as_string"], "%Y-%m-%d")
-        time_period = {"fiscal_year": str(key_as_date.year)}
+        time_period = {}
+        if self.group == "calendar_year":
+            time_period["calendar_year"] = str(key_as_date.year)
+        else:
+            time_period["fiscal_year"] = str(key_as_date.year)
+
+        # extract the categorical breakdown
+        categories_breakdown = bucket["group_by_category"]["buckets"]
+
+        # Initialize a dictionary to hold the query results for each obligation type
+        category_dictionary = {
+            "Contract_Obligations": 0,
+            "Direct_Obligations": 0,
+            "Grant_Obligations": 0,
+            "Idv_Obligations": 0,
+            "Loan_Obligations": 0,
+            "Other_Obligations": 0,
+        }
+
+        for category in categories_breakdown:
+            if category["key"] == "contract":
+                category_dictionary["Contract_Obligations"] = category.get("sum_as_dollars", {"value": 0})["value"]
+            elif category["key"] == "direct payment":
+                category_dictionary["Direct_Obligations"] = category.get("sum_as_dollars", {"value": 0})["value"]
+            elif category["key"] == "grant":
+                category_dictionary["Grant_Obligations"] = category.get("sum_as_dollars", {"value": 0})["value"]
+            elif category["key"] == "idv":
+                category_dictionary["Idv_Obligations"] = category.get("sum_as_dollars", {"value": 0})["value"]
+            elif category["key"] == "loans":
+                category_dictionary["Loan_Obligations"] = category.get("sum_as_dollars", {"value": 0})["value"]
+            elif category["key"] == "other":
+                category_dictionary["Other_Obligations"] += category.get("sum_as_dollars", {"value": 0})["value"]
+            elif category["key"] == "insurance":
+                category_dictionary["Other_Obligations"] += category.get("sum_as_dollars", {"value": 0})["value"]
 
         if self.group == "quarter":
```

Comment:
Can you move this block above the category breakdown logic. Would like to keep the code defining the `time_period` dictionary visually grouped together (even though it won't have any impact on the functionality)

Reply:
Yes I have moved the chunk of code with the code defining `time_period` and left a brief comment regarding the difference between `fiscal_year` and `calendar_year`